facade/service.go

```diff
@@ -1336,21 +1336,50 @@ func (f *Facade) ScheduleServiceBatch(ctx datastore.Context, svcs []*service.Ser
 
 func (f *Facade) updateDesiredState(ctx datastore.Context, svc *service.Service, desiredState service.DesiredState) error {
 	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.updateDesiredState"))
-	switch desiredState {
-	case service.SVCRestart:
-		// shutdown all service instances
-		if err := f.zzk.StopServiceInstances(ctx, svc.PoolID, svc.ID); err != nil {
+	if desiredState == service.SVCRestart {
+		//  place-holder ready func until we figure out how best to determine if we should move on to the next instance
+		if err := f.rollingRestart(ctx, svc, func() bool {
+			time.Sleep(5 * time.Second)
+			return true
+		}); err != nil {
+			glog.Errorf("Facade.updateDesiredState: Could not perform rolling restart for service %s (%s): %s", svc.Name, svc.ID, err)
 			return err
```

Comment:
Maybe we should fall back to a hard restart in this case.  If a host is offline and the user hits "restart", they won't get any feedback unless they look at the logs.  I don't know, its worth discussing.